### ISSUE TITLE
Stop training when a given loss value is reached

### ISSUE THREAD

**Issue by carlotardivo** [catboost/catboost#1862]
Hi, I wasn't able to find any way to stop the training by setting an upper bound over the validation set loss. For example, given a loss value the for me is ok (let's call it VAL) I'd like to stop the training once the loss value on the validation set is <= VAL. 

As a workaround I can for sure split the training into several steps  by saving the intermediate models and restart the training using the init_model argument. However this solution is suboptimal in terms of total number of performed iterations, since there will be always some extra iterations that have been done. Moreover -  but this could be the topic for another issue - it does not seem to me that training one model in one step gives the same identical results of training it in several steps as described above.

I think that this feature could be very useful, or - if a way to perform this already exists - a more exhaustive documentation could be helpful. 

Thanks.

catboost version: 0.25.1

**Comment by kizill** [comment]
Hello! In CatBoost 0.26 we released support for callbacks, so your task is easily. We are preparing our new documentation, as this function still is not described, but you can look at example in our tests:
https://github.com/catboost/catboost/blob/v0.26.1/catboost/python-package/ut/medium/test.py#L9726-L9738
I guess that's a good sample to start with.

**Comment by kizill** [comment]
Closing as solution is already available, feel free to reopen if something is not clear still.
